### PR TITLE
v3: vschema broken out by keyspace

### PR DESCRIPTION
**Pull request by sougou**
@alainjobart

---
This change is [Reviewable](https://reviewable.io/reviews/youtube/vitess/1615)

### CODE REVIEW

File: go/vt/topo/server.go [hunk 1]
```diff
@@ -312,12 +312,12 @@ type Impl interface {
 	//
 
 	// SaveVSchema saves the provided schema in the topo server.
-	SaveVSchema(context.Context, string) error
+	SaveVSchema(context.Context, string, string) error
```
Comment (alainjobart) [inline go/vt/topo/server.go]: Can you name the parameters, so we know what they are?

File: go/vt/vtgate/planner.go [hunk 2]
```diff
@@ -43,10 +52,53 @@ func NewPlanner(vschema *vindexes.VSchema, cacheSize int) *Planner {
 	return plr
 }
 
+// LoadVSchema loads the VSchema from the topo. The function does
+// not return an error. It instead logs warnings on failure.
+func (plr *Planner) LoadVSchema() {
+	formal := &vindexes.VSchemaFormal{
+		Keyspaces: make(map[string]vindexes.KeyspaceFormal),
+	}
+	keyspaces, err := plr.serv.GetSrvKeyspaceNames(context.TODO(), plr.cell)
```
Comment (alainjobart) [inline go/vt/vtgate/planner.go]: Usually you would pass in the Context into this function, let the higher levels create the one it's gonna use (that ends up being main() somewhere eventually, and then it's ok)

**Comment by alainjobart** [comment]
LGTM after (minor) comments are addressed.

File: go/vt/vtgate/planner.go (hunk 2 above)
Comment (sougou) [inline go/vt/vtgate/planner.go]: This is also the function you'd call when notified of a change, which is why I didn't add the context. Do you plan to create one in such cases?

Comment (alainjobart) [inline go/vt/vtgate/planner.go]: Yes, I would pass in the context (that the watcher will need anyway, and can be used to interrupt the watch).

I was planning on doing the watch per keyspace (and not global as you have it there), as an extra interface on the TopoServer, similar to WatchSrvKeyspace in go/vt/topo/server.go. We might need to wrestle with the code a bit to do that.

(these changes can and probably should be a different PR though, we can do that later, proceed with this).

Comment (sougou) [inline go/vt/vtgate/planner.go]: Yeah. The watcher should be per-keyspace. But the loader will need to re-read all vschemas to recompute dups. We could cache the vschemas, but it's not worth it.

**Pull request merged**